### PR TITLE
Merchant dashboard us36

### LINKED ISSUE #9: User Story 36, Merchant's Items index page
As a merchant employee
When I visit my merchant dashboard
I see a link to view my own items
When I click that link
My URI route should be "/merchant/items"

### PR DESCRIPTION
**Pull request by priyapower**
closes #9 
**Contributors**
- Driver(s): Priya
- Navigator(s): Luke
  
**What was changed:**
- Added merchant items view page with updated route, controller action, and link from dashboard page

**Any tech notes (if applicable)**
- Update view page for items view since it is only calling item.name right now

### CODE REVIEW

File: app/controllers/merchant/dashboard_controller.rb [hunk 1]
```diff
@@ -25,4 +25,8 @@ def index
     @pending_data = @pending_orders.zip(@pending_items.zip(@pending_item_orders))
   end
 
+  def items
+    @items = Item.where(merchant_id: current_user.merchant_id)
+  end
+
```
Comment (LHJE) [inline app/controllers/merchant/dashboard_controller.rb:31]: Glad we could just automatically repurpose code from above!

File: app/views/merchant/dashboard/index.html.erb [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+        <p><%= "Created: #{info.created_at}" %></p>
+      <% elsif info.class == ItemOrder %>
+        <p><%= "Total Quantity of Your Items: #{info.quantity}"  %></p>
+        <p><%= "Total Value: #{info.price * info.quantity}"  %></p>
+      <% end %>
     <% end %>
   <% end %>
-<% end %>
+
+</section>
```
Comment (LHJE) [inline app/views/merchant/dashboard/index.html.erb:23]: Smart move, adding it into a section.

File: app/views/merchant/dashboard/items.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<% @items.each do |item| %>
+  <%= item.name %>
+<% end  %>
```
Comment (LHJE) [inline app/views/merchant/dashboard/items.html.erb:3]: I like that we added this just for a more robust test. Diligent!

File: spec/features/merchant/show_spec.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 require 'rails_helper'
 
-RSpec.describe 'Site Navigation' do
+RSpec.describe 'Merchant Dashboard' do
```
Comment (LHJE) [inline spec/features/merchant/show_spec.rb:3]: Good catch

**Review by LHJE — APPROVED** [review]

File: app/controllers/merchant/dashboard_controller.rb (hunk 1 above)
Comment (priyapower) [reply app/controllers/merchant/dashboard_controller.rb:31]: Yes! I wonder if this means we can refactor the items action into the index action. Maybe that's not how this kind of code works....

File: app/views/merchant/dashboard/index.html.erb (hunk 2 above)
Comment (priyapower) [reply app/views/merchant/dashboard/index.html.erb:23]: Woo!

File: app/views/merchant/dashboard/items.html.erb (hunk 3 above)
Comment (priyapower) [reply app/views/merchant/dashboard/items.html.erb:3]: I wonder if more is coming for this page

**Pull request merged**